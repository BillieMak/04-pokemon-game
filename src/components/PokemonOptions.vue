<template>
    <div class="options-container">
        <ul>
            <li v-for="pokemon in pokemons" 
            :key="pokemon.id" 
            @click="$emit('selection', pokemon.id)">
            {{ pokemon.name }}
            </li>
        </ul>
    </div>
    <!-- <Button label="cuchi" class="dsd" severity="warning" raised icon="pi pi-plus-circle" iconPos="right" size="small" /> -->
</template>
<script>

import Button from 'primevue/button'

export default {
    name: 'PokemonOptions',
    props: {
        pokemons: {
            type: Array,
            required: true
        },
    },
    emits: ['selection'],
    components: {
        Button
    },
}

</script>

<style scoped>
.options-container{
    margin-top: 20px;
}
ul {
    list-style-type: none;
}

li {
    /* background-color: white; */
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    color: white;
    background-color: rgb(76, 135, 246);
    margin-bottom: 25px;
    width: 200px;
    padding: 10px 0;
    box-shadow: 0 0 10px 6px rgba(0, 0, 0, 0.2);
    text-transform: capitalize;
    font-size: 16px;
    font-weight: bold;
}

li:hover {
    background-color: rgba(18, 99, 248, 0.9);
}
li:active {
    transform: scale(0.9);
}

.options-container {
    display: flex;
    justify-content: center;
}
</style>