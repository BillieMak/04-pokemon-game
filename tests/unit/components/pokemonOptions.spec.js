import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { pokemonsMock } from "../mocks/pokemons.mock";

describe("Pruebas en PokemonOptions.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: pokemonsMock,
      },
    });
  });
  test("Debe de hacer match con el spanpshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe de mostrar las 4 opciones Correctamente", () => {
    const liTags = wrapper.findAll("li");
    expect(liTags).toHaveLength(4);

    expect(liTags.at(0).text()).toBe("bulbasaur");
    expect(liTags.at(1).text()).toBe("ivysaur");
    expect(liTags.at(2).text()).toBe("venusaur");
    expect(liTags.at(3).text()).toBe("charmander");
  });

  test('Debe de emitir "selection con sus respectivos parametros"', () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");
    li1.trigger("click");

    expect(wrapper.emitted().selection).toBeTruthy();
  });
});
