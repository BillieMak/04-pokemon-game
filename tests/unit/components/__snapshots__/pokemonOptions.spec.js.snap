// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PokemonOptions.vue Debe de hacer match con el spanpshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
<!-- <Button label="cuchi" class="dsd" severity="warning" raised icon="pi pi-plus-circle" iconPos="right" size="small" /> -->
`;
